internal/lines.go

```diff
@@ -61,6 +75,20 @@ func (lh *LineHandler) Flush() error {
 	return nil
 }
 
+func (lh *LineHandler) FlushAll() error {
+	lh.mtx.Lock()
+	defer lh.mtx.Unlock()
+	bufLen := len(lh.buffer)
+	if bufLen > 0 {
+		lines := make([]string, bufLen)
+		for i := 0; i < bufLen; i++ {
+			lines[i] = <-lh.buffer
+		}
+		return lh.report(lines)
```

Comment:
We should limit to the batch size even when we flush the entire buffer. We can delegate to Flush() until the buffer is drained. But bail out if there is an error when reporting to avoid an infinite loop as points are added back to buffer in that case.

Reply:
Actually, now that I think about it, why should Flush() even flush partially? Isn't it better if it flushes everything in batches until buffer is empty?